aiida/tools/importexport/common/archive.py

```diff
@@ -218,7 +220,23 @@ def _read_json_file(self, filename):
             return json.load(fhandle)
 
 
-def extract_zip(infile, folder, nodes_export_subfolder=None, silent=False):
+def update_description(path, progress_bar):
+    """Update description for a progress bar given path
+
+    :param path: path of file or directory
+    :type path: str
+
+    :param progress_bar: `tqdm` progress bar to have its description updated
+    """
+    (path, description) = os.path.split(path)
+    while description == '':
+        (path, description) = os.path.split(path)
+    description = 'EXTRACTING: {}'.format(description)
+
+    progress_bar.set_description_str(description)
```

Comment:
Note: by default this forces a refresh, see https://tqdm.github.io/docs/tqdm/#set_description_str
I don't think we want to do this hundreds of thousands of times.

Reply:
It doesn't seem to be a problem in my tests with big export archives - but I agree and will implement the `update` parameter in the `update_description()` utility function, with a default of `False`.